packages/api/src/beacon/client/debug.ts

```diff
@@ -1,60 +1,23 @@
 import {ChainForkConfig} from "@lodestar/config";
-import {ApiClientResponse, ResponseFormat} from "../../interfaces.js";
-import {HttpStatusCode} from "../../utils/client/httpStatusCode.js";
-import {generateGenericJsonClient, getFetchOptsSerializers, IHttpClient} from "../../utils/client/index.js";
-import {StateId} from "../routes/beacon/state.js";
-import {Api, getReqSerializers, getReturnTypes, ReqTypes, routesData} from "../routes/debug.js";
+import {ApiClientMethods, createApiClientMethods, IHttpClient} from "../../utils/client/index.js";
+import {definitions, Endpoints} from "../routes/debug.js";
 
 // As Jul 2022, it takes up to 3 mins to download states so make this 5 mins for reservation
 const GET_STATE_TIMEOUT_MS = 5 * 60 * 1000;
 
 /**
  * REST HTTP client for debug routes
  */
-export function getClient(_config: ChainForkConfig, httpClient: IHttpClient): Api {
-  const reqSerializers = getReqSerializers();
-  const returnTypes = getReturnTypes();
-  // Some routes return JSON, use a client auto-generator
-  const client = generateGenericJsonClient<Api, ReqTypes>(routesData, reqSerializers, returnTypes, httpClient);
-  // For `getState()` generate request serializer
-  const fetchOptsSerializers = getFetchOptsSerializers<Api, ReqTypes>(routesData, reqSerializers);
+export function getClient(_config: ChainForkConfig, httpClient: IHttpClient): ApiClientMethods<Endpoints> {
+  const client = createApiClientMethods(definitions, httpClient);
 
   return {
     ...client,
-
-    // TODO: Debug the type issue
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-expect-error
-    async getState(stateId: string, format?: ResponseFormat) {
-      if (format === "ssz") {
-        const res = await httpClient.arrayBuffer({
-          ...fetchOptsSerializers.getState(stateId, format),
-          timeoutMs: GET_STATE_TIMEOUT_MS,
-        });
-        return {
-          ok: true,
-          response: new Uint8Array(res.body),
-          status: res.status,
-        } as ApiClientResponse<{[HttpStatusCode.OK]: Uint8Array}>;
-      }
-      return client.getState(stateId, format);
+    getState(args, init) {
```

Comment:
Does the second argument are request options or initialization options?

Reply:
second arg is [`RequestInit`](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html)` & LodestarExtras`